x-pack/plugins/ml/public/maps/util.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FeatureCollection, Feature } from 'geojson';
+import { ESSearchResponse } from '../../../../../src/core/types/elasticsearch';
+import { formatHumanReadableDateTimeSeconds } from '../../common/util/date_utils';
+import type { MlApiServices } from '../application/services/ml_api_service';
+import { MLAnomalyDoc } from '../../common/types/anomalies';
+import type { SearchFilters } from './anomaly_source';
+
+export type MlAnomalyLayers = 'typical' | 'actual' | 'connected';
+
+export async function getResultsForJobId(
```

Comment:
I reckon instead of util function, it should be a proper service with `mlResultsService` injected as a dependency.

Reply:
I also think this method deserves a unit test

Reply:
Good call - added to follo-ups 👍